### ISSUE TITLE
Consider Google Colab as an alternative to MyBinder

### ISSUE THREAD

**Issue by jcohenadad** [jcohenadad/GBM8378#25]

**Comment by po09i** [comment]
The biggest problem seems to be that colab does not support creating environment for specific notebooks. I tried to remedy this by including some code in a cell but that's not ideal. It is possible to download and install dependencies from a cell but if we change python interpreter, we need to restart the kernel.

Now that we have fixed myBinder in #30, I don't think it's worth moving towards colab.

**Comment by jcohenadad** [comment]
> The biggest problem seems to be that colab does not support creating environment for specific notebooks.

When running a notebook under colab, a new environment _is_ created. Then, dependencies can be installed with pip, exactly like with MyBinder. Example:
~~~
!pip install tensorflow
~~~

If there are conda deps (which seems to be the case here), [here](https://medium.com/data-professor/how-to-install-conda-on-google-colab-e7bbf9036f76) are some instructions (disclaimer: I have not tested). 

But if the MyBinder option is fixed, I guess there is no need in pursuing the Google colab approach so we can lower the priority on this issue. The only problem, though is that it takes a very long time to load MyBinder (see https://github.com/jcohenadad/GBM8378/pull/30#pullrequestreview-1201072614), so I suggest we keep this issue open in case colab happens to be much faster.

**Comment by po09i** [comment]
Yes I tried similar approaches. The code you linked seems promising. I modified it so that is uses the latest python 3.6 version:
```
################################################################################
# INSTALL CONDA ON GOOGLE COLAB
################################################################################
! wget https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh
! chmod +x Miniconda3-4.5.4-Linux-x86_64.sh
! bash ./Miniconda3-4.5.4-Linux-x86_64.sh -b -f -p /usr/local
import sys
sys.path.append('/usr/local/lib/python3.6/site-packages/')
```

We would then need to install the dependencies:
- git clone then install a requirement file (The current file creates an environment, from what I know, that's not supported, we need to stick to the one created in base).
- put the dependencies directly in the cell.
